README.md

```diff
@@ -1,2 +1,177 @@
+
+## Mechanism
+
+When you write this in your code:
+```go
+    logger, _ := zap.NewProduction()
+```
+you are creating a `zap.Logger` instance which is a wrapper around `zapcore.Core` instance. 
+`zapcore.Core` is the actual `log backend` that implement a `Write` method that outputs a log records to some destination.
+
+This module provides a `zapcore.Core` implementation that initialize an OpenTelemetry logger SDK,
+which is then used to export log records to some destination with an opentelemetry exporter.
+
+## Configuration
+
+Currently the supported configuration method is via OpenTelemetry standard environment variables which you can find [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md) and [here for otlp exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md).
+
+Specifically, I found the following environment variables useful:
+- `OTEL_SERVICE_NAME` - to add a service name to the logs records resource, which is must-have for them to be useful.
+- `OTEL_SDK_DISABLED=true` - to disable the OpenTelemetry SDK, which allows a one switch to turn off otel SDK if needed (for example, local development, or as a step in migration)
+- `OTEL_EXPORTER_OTLP_PROTOCOL=grpc` - the default otlp protocol is http, which is less performant than grpc. this is how you can change it.
+- `OTEL_EXPORTER_OTLP_INSECURE=true` - **Only if you use internal OpenTelemetry collector gateway** set when you export your logs over insecure (not TLS) connection. It is recommended to run an OpenTelemetry collector in your cluster to server as a gateway to your log vendor or log backend which your use. This redouces the overhead on the application that do not need to encrypt and export to a local receiver in the cluster.
+- `OTEL_EXPORTER_OTLP_ENDPOINT`/`OTEL_EXPORTER_OTLP_LOGS_ENDPOINT` - the endpoint of the OpenTelemetry collector to which you want to export your logs. recommended: local OpenTelemetry collector gateway in your cluster. You can also use and otlp compatible receiver like you log vendor endpoints.
+- `OTEL_EXPORTER_OTLP_HEADERS` / `OTEL_EXPORTER_OTLP_LOGS_HEADERS` - If you use a vendor, you might get instructions to add some headers to your requests for authentication or other purposes. This is what you need to use for it to work.
+
+The rest of the configuration has reasonable defaults, but you can find browse through them to fine tune to your needs.
+
+## Usage
```

Comment:
I think that's the most interesting part, it maybe worth moving before the motivation section.